onnxruntime/core/session/inference_session.cc

```diff
@@ -474,31 +471,31 @@ common::Status InferenceSession::Load(std::unique_ptr<ModelProto> p_model_proto)
       AddCustomOpDomains({domain.get()});
     }
 #endif
-    return onnxruntime::Model::Load(std::move(p_model_proto), PathString(), model,
+    return onnxruntime::Model::Load(std::move(*p_model_proto), PathString(), model,
```

Comment:
The Model class has calls only from other ORT layers and we remove support for std::unique_ptr<ModelProto> in the static Model::Load() method. Instead we support ModelProto&& , hence we cast the l-value ref to r-value ref and pass on the ownership to the static Model::Load() method